llvm/lib/Transforms/Scalar/Reassociate.cpp

```diff
@@ -625,10 +532,7 @@ static bool LinearizeExprTree(Instruction *I,
       // Node initially thought to be a leaf wasn't.
       continue;
     assert(!isReassociableOp(V, Opcode) && "Shouldn't be a leaf!");
-    APInt Weight = It->second;
-    if (Weight.isMinValue())
-      // Leaf already output or weight reduction eliminated it.
-      continue;
+    uint64_t Weight = It->second;
```

Comment:
As we don't reduce weight for nilpotent ops now (e.g., xor), the weight cannot be zero.